algosdk/future/transaction.py

```diff
@@ -2906,6 +2909,103 @@ def __eq__(self, other):
         return False
 
 
+class StateProofTxn(Transaction):
+    """
+    Represents a state proof transaction
+
+    Arguments:
+        sender (str): address of the sender
+        state_proof (dict())
+        state_proof_message (dict(), optional)
+        state_proof_type (str, optional): state proof type
+        sp (SuggestedParams): suggested params from algod
+        receiver (str): address of the receiver
```

Comment:
Basically, a state proof transaction should not have any fee, note, rekeying field, or lease. I don't think we should assert that, but maybe just change the comments?